docs/development/testguide.rst

```diff
@@ -25,6 +25,105 @@ accessed through the ``python setup.py test`` command.
     see the two spellings used interchangeably in the documentation.
 
+.. _testing-dependencies:
+
+Testing Dependencies
+********************
+
+As of Astropy 3.0, the dependencies used by the Astropy test runner are
+provided by a separate package called ``pytest-astropy``. This package provides
+the ``pytest`` dependency itself, in addition to several ``pytest`` plugins
+that are used by Astropy, and will also be of general use to other packages.
+
+Since the testing dependencies are not actually required to install or use
+Astropy, they are not included in ``install_requires`` in ``setup.py``.
+However, for technical reasons it is not currently possible to express these
+dependencies in ``tests_require`` either. Therefore, ``pytest-astropy`` is
+listed as an extra dependency using ``extras_require`` in ``setup.py``.
+Developers who want to run the test suite will need to install the testing
+package using pip::
+
+    > pip install pytest-astropy
+
+The following ``pytest`` plugins are included in the ``pytest-astropy``
+package:
+
+pytest-remotedata
+=================
+
+The `pytest-remotedata`_ plugin allows developers to control whether to run
+tests that access data from the internet. The plugin provides two decorators
+that can be used to mark individual test functions or entire test classes:
+
+* ``@pytest.mark.remote_data`` for tests that require data from the internet
+* ``@pytest.mark.internet_off`` for tests that should run only when there is no
+  internet access. This is useful for testing local data caches or fallbacks
+  for when no network access is available.
+
+The plugin also adds the ``--remote-data`` option to the ``pytest`` command
+(which is also made available through the Astropy test runner).
+
+The default behavior is to skip tests that are marked with ``remote_data``. If
+the ``--remote-data`` option is not provided when running the test suite, or if
+``--remote-data=none`` is provided, all tests that are marked with
+``remote_data`` will be skipped. All tests that are marked with
+``internet_off`` will be executed. Any test that attempts to access the
+internet but is not marked with ``remote_data`` will result in a failure.
+
+Providing either the ``--remote-data`` option, or ``--remote-data=any``, will
+cause all tests marked with ``remote_data`` to be executed. Any tests that are
+marked with ``internet_off`` will be skipped.
+
+Running the tests with ``--remote-data=astropy`` will cause only tests that
+receive remote data from Astropy data sources to be run. Tests with any other
+data sources will be skipped. This is indicated in the test code by marking
+test functions with ``@pytest.mark.remote_data(source='astropy')``.
+
+Also see :ref:`data-files`.
+
+.. _pytest-remotedata: https://github.com/astropy/pytest-remotedata
+
+pytest-doctestplus
+==================
+
+The `pytest-doctestplus`_ plugin provides advanced doctest features, including:
```

Comment:
Nitpick: Do we need to use double-backticks around doctest to render it as `doctest` since "doctest" is not a real word? I'll leave it to the linguist to decide.

Reply:
Hmm, personally I think it's common enough `term` in `python` circles to not need `backticks` :smile: